src/envConfig.ts

```diff
@@ -2,54 +2,60 @@ import fs from 'fs';
 import path from 'path';
 import type { Plugin } from 'vite';
 
-/**
- * This plugin serves the `/assets/env-config.js` file with proper environment variables
- */
-
-export function renderTemplateWithEnvVars(content: string): string {
-  let result = content;
-  const regex = /\$\{([a-zA-Z_]+[a-zA-Z0-9_]*?)\}/g;
-
-  let matched: RegExpExecArray | null;
-  while ((matched = regex.exec(content))) {
-    result = result.replace(`\${${matched[1]}}`, process.env[matched[1]] || '');
-  }
+export function createEnvConfigContent(variables: string[], template: boolean): string {
+  let templateContent = '';
+  templateContent += '(function(window){window.env=window.env||{};';
+  variables.forEach((variable) => {
+    if (template) {
+      templateContent += `window.env['${variable}']='\${${variable}}';`;
+    } else {
+      templateContent += `window.env['${variable}']='${process.env[variable] || ''}';`;
+    }
+  });
+  templateContent += '})(this);';
+  return templateContent;
+}
 
-  return result;
+export interface EnvConfigOptions {
+  variables: string[];
 }
 
-export function envConfig(): Plugin {
-  let root: string | undefined;
+export function envConfig(userOptions: Partial<EnvConfigOptions> = {}): Plugin {
+  let root: string;
   return {
     name: 'vite-plugin-env-config',
+
     configResolved(config) {
       root = config.root;
     },
-    configureServer(server) {
-      if (!root) {
-        throw new Error('envConfig plugin requires a root directory');
-      }
-      let appConfigContent = '';
-      const TEMPLATE_PATH = path.join(root, 'src', 'assets', 'env-config.template.js');
 
-      void (async () => {
-        const content = (await fs.promises.readFile(TEMPLATE_PATH)).toString();
-        appConfigContent = renderTemplateWithEnvVars(content);
-      })();
+    configureServer(server) {
+      const envConfigContent = createEnvConfigContent(userOptions.variables || [], false);
```

Comment:
```suggestion
      const envConfigContent = createEnvConfigContent(userOptions.variables.map(s => s.toUpperCase()) || [], false);
```

Reply:
@anbraten Can't environment variables be lower case?

Reply:
We can leave it as is for now.